curator-documentation/OCP-Metering-Options.md

```diff
@@ -0,0 +1,99 @@
+
+###### 2. Tectonic- This is a container as a service tool that manages containers and deploys application
+*   Tectonic Chargeback- aggregates Prometheus data and generates reports based on the collected usage information
+*   Only available in admin console on OpenShift (cannot find a lot of information on this
+
+# More on Koku:
+
+The following is setup instructions/tech requirements for koku backend based on provided documentation. It is quite extensive as a setup.
+
+1. Make sure project is running OCP version 3.11 or newer.
+2. Install metering operator (https://github.com/operator-framework/operator-metering/blob/master/Documentation/install-metering.md) on your OCP cluster.
+3. Expose a route (https://github.com/operator-framework/operator-metering/blob/master/Documentation/configuring-reporting-operator.md#openshift-route) for the reporting operator.
+4. Install and configure the [Red Hat Insights Client](https://access.redhat.com/products/red-hat-insights/#getstarted) on a system with network access to your OCP cluster.
+5. Install [Ansible](https://docs.ansible.com/ansible/2.7/installation_guide/intro_installation.html) and the [EPEL repository](https://fedoraproject.org/wiki/EPEL#Quickstart) on the system where the Red Hat Insight Client is installed.
+6. [Install the OCP command line, oc](https://docs.openshift.com/container-platform/3.3/cli_reference/get_started_cli.html#cli-linux), on the system where the Red Hat Insights Client is installed.
+7. They have an Ansible playbook called OCP Usage Collector that needs to be setup as well
+8. OCP usage collector retrieves the data from Metering Operator endpoint (similar to our back end)
+
+
+# Koku UI
+*   Koku is the upstream project for Red Hat OpenShift Container Platform cost management [https://cloud.redhat.com/cost-management/](https://cloud.redhat.com/cost-management/).
+*   At the moment, UI observations are taken from screenshots online since Koku UI requires insights-proxy setup as well as a few DNS config changes that seem to hang when UI starts.
+
+Observations:
+*   All metrics are in Cost
+*   There is a “top projects”, “top clusters”, “top nodes” component so user can easily see where resources are being used the most.
+*   Under cost management there is an overview page as well as a details page
+
+*   The details page is similar to our reports page and the columns are:
+    *   Project Name
+    *   Month over month change (so has usage increased or decreased from the previous month)
+    *   Cost model values
+    *   Cost change
+
+**For  frontend they have 4 different api calls on the UI that correspond to their API design:**
+
+```
+[ReportType.cost]: 'reports/openshift/costs/',
+[ReportType.cpu]: 'reports/openshift/compute/',
+[ReportType.memory]: 'reports/openshift/memory/',
+[ReportType.volume]: 'reports/openshift/volumes/'
+```
+The four values are what are utilized in metrics displayed on the UI.
+
```

Comment:
@shruthi-raghuraman Can you please add more information about the purpose of these API calls? Also, we need to make sure that we aren't considering cost as an attribute in any of our research.